addons/inventory/XEH_postInit.sqf

```diff
@@ -9,46 +9,22 @@ if (!hasInterface) exitWith {};
 // luckily we don't need private items, so dummy and parent classes are out of the picture
 
 GVAR(ItemKeyNamespace) = [] call CBA_fnc_createNamespace;
+private _allItems = uiNamespace getVariable [QGVAR(ItemKeyCache), []]; //See XEH_preStart.sqf
 
-private _fnc_addToCache = {
-    private _displayName = getText (_this >> "displayName");
-    private _picture = getText (_this >> "picture");
-
-    // list box seems to delete the leading backslash
-    if (_picture select [0,1] == "\") then {
-        _picture = _picture select [1];
+//isEqualType is hacking protection as we cannot trust that the cache hasn't been manipulated
+{
+    if (_x isEqualType []) then {
```

Comment:
```suggestion
    if (_x isEqualType [] && {_x isEqualTypeArray ["", configNull]}) then {
```

Reply:
I found that this thing is the new major performance waste in this function.. 
Maybe compileFinal will be better in the end.

Reply:
Sounds simpler anyway.

Reply:
Nvm doesn't work.. Would need to re-resolve all the config entries which would be even worse